stingray/gti.py

```diff
@@ -393,3 +396,73 @@ def append_gtis(gti0, gti1):
             'exclusive.')
 
     return np.concatenate([gti0, gti1])
+
+
+def join_gtis(gti0, gti1):
+    """Union of two GTIs, general case
+
+    Parameters
+    ----------
+    gti0: 2-d float array
+        [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+
+    gti1: 2-d float array
+        [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+
+    Returns
+    -------
+    gti: 2-d float array
+        The newly created GTI
+    """
+
+    gti0 = np.asarray(gti0)
+    gti1 = np.asarray(gti1)
+
+    # Check if independently GTIs are well behaved.
+    check_gtis(gti0)
+    check_gtis(gti1)
+
+    # Check if GTIs are mutually exclusive. If so, append them
+    if check_separate(gti0, gti1):
+        return append_gtis(gti0, gti1)
+
+    # Otherwise, do a trickier procedure.
+    # When two GTIs overlap partially, we put them on a line and look at the
+    # number of opened and closed intervals. When the number of closed and
+    # opened intervals is the same, the full GTI is complete and we close it.
+    # In practice, we assign to each opening time of a GTI the value -1, and
+    # the value 1 to each closing time; when the cumulative sum is zero, the
+    # GTI has ended. The timestamp after each closed GTI is the start of a new
+    # one.
+
```

Comment:
I feel like these comments, along with the one above  ("check if GTIs are mutually exclusive ...") would be well placed in the docstring rather than the code. This is information I think a user might want to know.
Not sure what you mean by "opened and closed intervals", though.